### ISSUE TITLE
finetune ip-adapter-full-face_sd15.bin model questions 

### ISSUE THREAD

**Issue by chuck-ma** [tencent-ailab/IP-Adapter#154]
I'm currently working on finetuning ip-adapter-full-face_sd15.bin model.
Here are some questions:

1. How many face images should be used to finetune?
2. When the loss value drops to what level, it can be regarded as convergence ?
3. Besides, i simply switch ImageProj to a MLP in tutorial_train_plus.py like below:

`image_proj_model = MLPProjModel(
        cross_attention_dim=unet.config.cross_attention_dim,
        clip_embeddings_dim=image_encoder.config.hidden_size,
    )`
    
    And num_tokens change to 257. Is that will be ok?
    
    Any help will be appreciated.

**Comment by xiaohu2015** [comment]
1. I used about 50W images to train the model
2. you can sample examples during your training 
3. it is ok

**Comment by chuck-ma** [comment]
Besides, it seems that ip-adapter-full-face_sd15.bin works not well when inpainting.  Is there anything missing ?

Here is the code:
`image = Image.open("assets/images/ai_face2.png")
image.resize((256, 256))

masked_image = Image.open("/kaggle/input/beauty/beauty-1.png").resize((512, 512)).convert("RGB")
mask = Image.open("/kaggle/input/beauty/beauty-1-mask-2.png").resize((512, 512)).convert("RGB")
image_grid([masked_image.resize((256, 256)), mask.resize((256, 256))], 1, 2)

base_model_path = "runwayml/stable-diffusion-v1-5"
vae_model_path = "stabilityai/sd-vae-ft-mse"
image_encoder_path = "models/image_encoder/"
ip_ckpt = "models/ip-adapter-full-face_sd15.bin"
device = "cuda"

def image_grid(imgs, rows, cols):
    assert len(imgs) == rows*cols

    w, h = imgs[0].size
    grid = Image.new('RGB', size=(cols*w, rows*h))
    grid_w, grid_h = grid.size
    
    for i, img in enumerate(imgs):
        grid.paste(img, box=(i%cols*w, i//cols*h))
    return grid

noise_scheduler = DDIMScheduler(
    num_train_timesteps=1000,
    beta_start=0.00085,
    beta_end=0.012,
    beta_schedule="scaled_linear",
    clip_sample=False,
    set_alpha_to_one=False,
    steps_offset=1,
)
vae = AutoencoderKL.from_pretrained(vae_model_path).to(dtype=torch.float16)

pipe = StableDiffusionInpaintPipelineLegacy.from_pretrained(
    base_model_path,
    torch_dtype=torch.float16,
    scheduler=noise_scheduler,
    vae=vae,
    feature_extractor=None,
    safety_checker=None
)

ip_model = IPAdapterFull(pipe, image_encoder_path, ip_ckpt, device, num_tokens=257)
images = ip_model.generate(pil_image=image, num_samples=4, num_inference_steps=50,
                           seed=42, image=masked_image, mask_image=mask, strength=0.99, )
grid = image_grid(images, 1, 4)
grid


`
here is result 



Here is origin image



Here is masked image and mask:
